src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/PrisonPersonMoveBookingService.kt

```diff
@@ -0,0 +1,80 @@
+package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.prisonperson
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.config.trackEvent
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.data.PrisonerBookingMovedDomainEvent
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.prisonperson.physicalattributes.PhysicalAttributesNomisApiService
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.prisonperson.physicalattributes.PhysicalAttributesSyncService
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.prisonperson.physicalattributes.findLastModifiedPhysicalAttributes
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.prisonperson.physicalattributes.lastModifiedDateTime
+
+@Service
+class PrisonPersonMoveBookingService(
+  private val telemetryClient: TelemetryClient,
+  private val physicalAttributesSyncService: PhysicalAttributesSyncService,
+  private val physicalAttributesNomisApiService: PhysicalAttributesNomisApiService,
+  private val prisonPersonNomisSyncApiService: PrisonPersonNomisSyncApiService,
+) {
+
+  suspend fun bookingMoved(bookingMovedEvent: PrisonerBookingMovedDomainEvent) {
+    val bookingId = bookingMovedEvent.additionalInformation.bookingId
+    val toOffenderNo = bookingMovedEvent.additionalInformation.movedToNomsNumber
+    val fromOffenderNo = bookingMovedEvent.additionalInformation.movedFromNomsNumber
+    val telemetry = mutableMapOf(
+      "bookingId" to bookingId.toString(),
+      "toOffenderNo" to toOffenderNo,
+      "fromOffenderNo" to fromOffenderNo,
+    )
+
+    try {
+      syncFromOffenderDps(fromOffenderNo, telemetry)
+
+      syncToOffenderDps(toOffenderNo, telemetry)
+
+      syncToOffenderNomis(toOffenderNo, telemetry)
```

Comment:
This is where the action happens...

- sync the "from offender" to DPS - DPS will have been updated with wrong values from the to offender's booking
- sync "to offender" to DPS where a value has changed - DPS retains the values from the previous booking, so needs a sync for any fields that were updated
- sync "to offender" back to NOMIS - the NOMIS booking that was moved had incorrect values copied from the "from offender" so udpate them